samples/will_it_shuffle.lobster

```diff
@@ -0,0 +1,228 @@
+// adaptation of https://bost.ocks.org/mike/shuffle/compare.html
+//
+// TODO
+// - vectorize the b-spline
+// - more shufflers
+// - measure reshuffle time
+// - swappable RNG
+
+import color
+import imgui
+import std
+import vec
+
+enum shuffle_mode:
+    shuffle_none = 0
+    shuffle_naive_swap
+    shuffle_push_remove
+    shuffle_fisher_yates
+
+var mode = shuffle_none
+
+def shuffle(array): switch mode:
+    case shuffle_none: array
+    case shuffle_naive_swap: naive_swap(array)
+    case shuffle_push_remove: push_remove(array)
+    case shuffle_fisher_yates: fisher_yates(array)
+
+def mode_name(): return switch mode:
+    case shuffle_none: "none"
+    case shuffle_naive_swap: "naive swap"
+    case shuffle_push_remove: "push remove"
+    case shuffle_fisher_yates: "fisher yates"
+    default: ""
+
+def naive_swap(xs):
+    // ala https://blog.codinghorror.com/the-danger-of-naivete/
+    for(xs) x, i:
+        let j = rnd(xs.length)
+        xs[i] = xs[j]
+        xs[j] = x
+
+def push_remove(xs):
+    // std.lobster original
+    for(xs.length) i:
+        xs.push(xs.remove(rnd(xs.length - i)))
+
+def fisher_yates(xs):
```

Comment:
use vs in `std` if that gets merged

Reply:
Simplified to match, but leaving both dual implemented here for clarity of the demo fwiw